docs/naps/9-multiple-canvases.md

```diff
@@ -0,0 +1,261 @@
+    * Just a list (`EventedList`) of `CanvasModel` objects
+    * One canvas is “active”, relevant properties on ViewerModel (`camera`, `dims`) are delegated to the active canvas
+        * "active" canvas is just index 0 (alternatively: use `SelectableEventedList`)
+    * Only the active canvas is shown (minimal modifications to `QtViewer` and `VispyCanvas`)
+    * A single `QtDims` view (slicing sliders) is shown, updates depending on the active canvas
+    * Adjust event callbacks such that interactions (slicing, camera movement, ndisplay toggle) only apply to the active canvas
+    * Add public APIs to add/remove/rotate (change active) `_canvases`
+    * *Note: An improved version of "Grid Mode" may be able to be implemented with just these changes.*
+
+2. Add Layer Slice classes to reduce data reslicing when switching between canvases
+    * Start with `Image` and `Labels` layers
+        * Move/modify/replace `set_view_slice` and `_update_slice_response` to set data on Layer Slice for associated canvas
+    * Modify `VispyCanvas`
+        * Get camera and dims information from associated `CanvasModel` instead of `Viewer`
+        * Obtain relevant data from Layer Slice instead of Layer
+    * Modify `ViewerModel` and `LayerSlicer` [^async-only]
+        * Submit `CanvasModel` (or an ID) to `LayerSlicer` instead of dims directly
+        * Emit `CanvasModel` (or an ID) and slice data from `LayerSlicer.ready` and/or `Layer.set_data` events
+        * `VispyCanvas` will subscribe to relevant events, set data if corresponding to its own `CanvasModel`. Other canvases may also be interested in this event for example to update cross-reference overlays.
+
+3. Update `QtViewer` and `VispyCanvas` to support multiple canvases
+    * Still only displaying one canvas at a time in the main widget
+    * Update main widget as `ViewerModel` “active” canvas changes, storing additional canvases and swapping them out as necessary
+
+4. Update `QtViewer` to show multiple canvases simultaneously
+    * This is exploratory work at the moment, see [UI Architecture](#UI-Design-and-Architecture) section below
+
+[^async-only]: Depending on the timeline and prototype implementation, it may be acceptable/preferable for multi-canvas feature to rely on (currently experimental) async slicing ([see NAP-4](https://napari.org/stable/naps/4-async-slicing.html)).
+
+## Backward Compatibility
+
+Maintaining the proxy API on the viewer via the concepts of a main and/or active canvas will make this work mostly backward-compatible, though there will inevitably be some breaking changes. There will like be significant breaking changes to private APIs. For example if plugins are attempting to access slice data directly from a layer instance, it may no longer be as expected. If this is a large burden, it too may be mitigated by delegating from the layer to the slice corresponding to the main or active canvas.
```

Comment:
```suggestion
Maintaining the proxy API on the viewer via the concepts of a main and/or active canvas will make this work mostly backward-compatible, though there will inevitably be some breaking changes. There will likely be significant breaking changes to private APIs. For example if plugins are attempting to access slice data directly from a layer instance, it may no longer be as expected. If this is a large burden, it too may be mitigated by delegating from the layer to the slice corresponding to the main or active canvas.
```